public/examples/ts/bar-drilldown.ts

```diff
@@ -1,114 +1,186 @@
+        divideShape: 'clone'
+      }
+    },
+    graphic: [
       {
-        value: 4,
-        groupId: 'cars'
+        type: 'text',
+        left: 50,
+        top: 20,
+        style: {
+          text: 'Back',
+          fontSize: 18,
+          fill: 'grey'
+        },
+        onclick: function () {
+          goBack();
+        }
       }
-    ] as DataItem[],
-    universalTransition: {
-      enabled: true,
-      divideShape: 'clone'
-    }
-  }
+    ]
+  };
+  allOptions[optionId] = option;
+});
+
+// A stack to remember previous option id
+const optionStack: string[] = [];
+
+const goForward = (optionId: string) => {
+  optionStack.push(myChart.getOption().id as string); // push current option id into stack.
+  myChart.setOption(allOptions[optionId]);
 };
 
-const drilldownData = [
-  {
-    dataGroupId: 'animals',
-    data: [
-      ['Cats', 4],
-      ['Dogs', 2],
-      ['Cows', 1],
-      ['Sheep', 2],
-      ['Pigs', 1]
-    ]
-  },
-  {
-    dataGroupId: 'fruits',
-    data: [
-      ['Apples', 4],
-      ['Oranges', 2]
-    ]
-  },
-  {
-    dataGroupId: 'cars',
-    data: [
-      ['Toyota', 4],
-      ['Opel', 2],
-      ['Volkswagen', 2]
-    ]
+const goBack = () => {
+  if (optionStack.length === 0) {
+    console.log('Already in root level!');
+  } else {
+    console.log('Go back to previous level.');
+    myChart.setOption(allOptions[optionStack.pop() as string]);
   }
-];
+};
 
-myChart.on('click', function (event) {
-  if (event.data) {
-    var subData = drilldownData.find(function (data) {
-      return data.dataGroupId === (event.data as DataItem).groupId;
-    });
-    if (!subData) {
-      return;
-    }
-    myChart.setOption<echarts.EChartsOption>({
-      xAxis: {
-        data: subData.data.map(function (item) {
-          return item[0];
-        })
-      },
-      series: {
-        type: 'bar',
-        id: 'sales',
-        dataGroupId: subData.dataGroupId,
-        data: subData.data.map(function (item) {
-          return item[1];
-        }),
-        universalTransition: {
-          enabled: true,
-          divideShape: 'clone'
-        }
-      },
-      graphic: [
-        {
-          type: 'text',
-          left: 50,
-          top: 20,
-          style: {
-            text: 'Back',
-            fontSize: 18
-          },
-          onclick: function () {
-            myChart.setOption<echarts.EChartsOption>(option);
-          }
-        }
-      ]
-    });
+option = allOptions['things']; // The initial option is the root data option
+
+myChart.on('click', 'series', (params) => {
+  const dataItem = params.data as any;
+  if (dataItem[3]) {
+    // If current params is not belong to the "childest" data, it has data[3]
+    const childGroupId = dataItem[3];
+    // since we use groupId as optionId in this example,
+    // we use childGroupId as the next level optionId.
+    const nextOptionId = childGroupId;
+    goForward(nextOptionId);
   }
 });
+
+option && myChart.setOption(option);
```

Comment:
Another question:

Changing the present example for bar drilldown will cause it can't work in versions <= 5.5.0. Maybe we should add a new one for multi-level drilldown?

Reply:
Hi @plainheart, thank you for review this PR.

> As an example, this line can be removed.

Got it. I will push another commit to this branch.

> Changing the present example for bar drilldown will cause it can't work in versions <= 5.5.0. Maybe we should add a new one for multi-level drilldown?

You are right, I missed that. So I should do:

1. keep the old drilldown example as what it was (it will continuely works since the multi-drilldown feature is compatible)
2. add a new multi-drilldown example

Any instruction on creating a new example? Should I follow any convention or idiom?

Reply:
It should be enough to create a new file `bar-multi-drilldown.ts` with your modified example code in `public/examples/ts`. Please keep the code concise and indicate this example requires v5.5.0 or later.

Finally, please run `npm run build:examplelist` to update the example list.

Reply:
Just found the present example for bar drilldown doesn't have a dynamic thumbnail, it would be great to tweak the code and enable auto-update before generating the thumbnails.

Reply:
Hi @plainheart, I need your help:

1. After building the `bar-multi-drilldown` example, the thumb is not correct since at the moment the example only works with local built echarts.js. How you guys handle this kind of situation when you have an example shows how to use a new feature but that feature is not available publicly yet?
<img width="572" alt="image" src="https://user-images.githubusercontent.com/32434520/205220775-fb42e96b-3037-4943-9c41-021aaed8dbe3.png">
<img width="569" alt="image" src="https://user-images.githubusercontent.com/32434520/205220938-ee50d92e-1390-4c94-88b7-bb8ec910e342.png">

2. Where should I put "this example requires v5.5.0 or later"? Is it supposed to be a comment in the start of `bar-multi-drilldown.ts` file? Could you please provide an example for me to learn?

3. What is a dynamic thumbnail? Is it a thumbnail as a gif file? I find there are parameters like `videoStart` and `videoEnd`, but examples with these two parameters will be played automatically, while at the moment `bar-drilldown` and `bar-multi-drilldown` don't. Do you mean this by saying "enable auto-update"?

Looking forward to your kind reply :-)

Reply:
1. You may use the `next` branch of echarts nightly. Just change [this line](https://github.com/apache/echarts-examples/blob/gh-pages/tool/screenshot.html#L23) to `https://cdn.jsdelivr.net/npm/echarts-nightly@next/dist/echarts.js`

2. Yep. Just add a comment like `This example requires ECharts v5.5.0 or later` at the beginning of the example code

3. It seems not easy to make a dynamic thumbnail for these two examples. It's okay not to do it for now. :)

Reply:
Your reply clear my mind, thank you so much!